babel/dates.py

```diff
@@ -1142,6 +1328,13 @@ def get_week_number(self, day_of_period, day_of_week=None):
     'z': [1, 2, 3, 4], 'Z': [1, 2, 3, 4], 'v': [1, 4], 'V': [1, 4]  # zone
 }
 
+#: The pattern characters declared in the Date Field Symbol Table
+#: (http://www.unicode.org/reports/tr35/tr35-dates.html#Date_Field_Symbol_Table)
+#: in order of decreasing magnitude.
+PATTERN_CHAR_ORDER = "GyYuUQqMLlwWdDFgEecabBChHKkjJmsSAzZvV"
+
+_pattern_cache = {}
```

Comment:
Consider creating a memoization decorator for this. The decorator could come in useful later when someone picks up caching parsed Locales. :)